src/core/border_router/routing_manager.cpp

```diff
@@ -888,7 +893,7 @@ bool RoutingManager::IsValidBrUlaPrefix(const Ip6::Prefix &aBrUlaPrefix)
 bool RoutingManager::IsValidOmrPrefix(const NetworkData::OnMeshPrefixConfig &aOnMeshPrefixConfig)
 {
     return IsValidOmrPrefix(aOnMeshPrefixConfig.GetPrefix()) && aOnMeshPrefixConfig.mOnMesh &&
-           aOnMeshPrefixConfig.mSlaac && aOnMeshPrefixConfig.mStable && !aOnMeshPrefixConfig.mDp;
+           aOnMeshPrefixConfig.mSlaac && aOnMeshPrefixConfig.mStable && aOnMeshPrefixConfig.mPreferred;
```

Comment:
Domian prefix may not have SLAAC? 

BTW, why excluding `mPreferred=false` ?

Reply:
> Domian prefix may not have SLAAC?

Yes. I guess then we won't accept/treat it as OMR. The main change is that we don't exclude it. 

> BTW, why excluding mPreferred=false ?

I think we can add this check directly here, but added it back in the code for now. May be better to change this in its own PR later. 
- Per spec 9.6.1.2 we need `mPreferred=true` for valid OMR. 
- When we are determining favored OMR prefix we ignore the ones that are `mPreferred== fasle`.
- I think it may be ok to add check in `IsValidOmrPrefix()` and do the same anywhere else we use this method.

Reply:
> > Domian prefix may not have SLAAC?
> 
> Yes. I guess then we won't accept/treat it as OMR. The main change is that we don't exclude it.
> 
> > BTW, why excluding mPreferred=false ?

My point was that a Domain Prefix tend to have `mSlaac = false`. But the current check still requires `aOnMeshPrefixConfig.mSlaac = true`, which is incorrect.

According to spec, a Domain Prefix should **NOT** have SLAAC enabled, but it's not a MUST.

> I think we can add this check directly here, but added it back in the code for now. May be better to change this in its own PR later.
> 
> * Per spec 9.6.1.2 we need `mPreferred=true` for valid OMR.
> * When we are determining favored OMR prefix we ignore the ones that are `mPreferred== fasle`.
> * I think it may be ok to add check in `IsValidOmrPrefix()` and do the same anywhere else we use this method.

We need to make sure to publish `mPreferred=false` OMR to RIO. Maybe better to use a separate PR for these change.

Reply:
> My point was that a Domain Prefix tend to have mSlaac = false. But the current check still requires aOnMeshPrefixConfig.mSlaac = true, which is incorrect.

Good point. I think this may be tricky and may require more thoughts. 

I think the idea for SLAAC is that we know all nodes can get an address with this prefix if they want to. Do we have the same with domain prefix as well (without SLAAC flag?) Do we need DUA feature to be enabled on all devices?

> We need to make sure to publish mPreferred=false OMR to RIO.

I see. Thanks. If we need this then I think we should keep the `IsValidOmrPrefix()` as it is (not check `mPreferred`).

Reply:
> I think the idea for SLAAC is that we know all nodes can get an address with this prefix if they want to. 

Agree

> Do we have the same with domain prefix as well (without SLAAC flag?) Do we need DUA feature to be enabled on all devices?

Thread 1.2 devices can self-generate DUA address and register to PBBR no matter the Domain Prefix has SLAAC flag or not. See spec 5.23.2.1 for more details.

Reply:
I guess for this the nodes need to enable `OPENTHREAD_CONFIG_DUA_ENABLE` feature, correct?
Is this a required feature for all nodes in a Thread 1.3 deployment? (I am not sure it is).

Reply:
DUA support is currently a MUST for any Thread 1.3 device. This follows from the DUA tests in the 1.2 certification, which 1.3 inherits. DUA support in a BR role (acting as ND-Proxy) is required if the BR supports configuration of a Domain Prefix, and not required if the BR can't support configuration of a Domain Prefix.  We had quite some discussions on this in the past and so far DUA never got removed :) 

Indeed a Domain Prefix with P_slaac = 0 is used by 1.2 or higher devices to create a Domain Address using SLAAC. However Thread 1.1 devices are left in the dark if P_slaac = 0 (they can't use SLAAC then), so this could be a reason for a BR to still create a new OMR prefix. On the other hand one could say that if a Domain Prefix is configured, the admin would have also configured another prefix manually to for use by 1.1 devices if any. So I tend to just simply let the BR assume that if P_dp = 1, it implies SLAAC is possible using that prefix regardless of P_slaac's value.

Reply:
I think the main question becomes:
- On BR (when deciding about OMR), if we see a prefix entry in net data with `mDp = 1` but `mSlaac=0`, do we consider it as valid OMR (and not add our local OMR) or not?  

My suggestion is no we don't consider it valid.

Reply:
There seems some issue for DUA prefix since it's not always OFF mesh reachable:
- DUA prefix with `mSlaac=0` can't address 1.1 devices
- DUA prefix with `mSlaac=1` can address 1.1 devices, but these DUAs are not registered, thus can't be properly ND proxyed

So, if we say `mSlaac=0` DUA prefix should not be considered as OMR, then neither should `mSlaac=1` DUA prefix, since they are both problematic. So, I think we should choose either 1. all DUA prefixes are OMR, 2. DUA prefix is never OMR.

Reply:
From another perspective, a 1.1 device can't register its address to SRP server thus can't be discovered from Backbone link. 
Maybe its reachability from off mesh does not matter in practise.

Reply:
Every solution has pros and cons, but I would opt to stay with the current spec text 9.6.1.2. This implies a prefix entry in net data with mDp = 1 but mSlaac=0 is not a valid OMR prefix, while mDp = 1 but mSlaac=1 is valid if the other conditions are also met.  (So @abtink 's choice)

If an operator configures a Domain Prefix and adds 1.1 devices (in the TC Commercial we would consider this as a corner case), the operator can choose between the two resulting behaviors depending on setting mSlaac = 0 / 1. Most likely in this scenario the operator is also able to configure the automatic OMR prefix configuration of BRs to either 'on' or 'off'.

If an operator configures a Domain Prefix and adds >= 1.2 devices (more likely scenario) then mSlaac could be to '1' and there would be no issue.

Reply:
> Every solution has pros and cons, but I would opt to stay with the current spec text 9.6.1.2. This implies a prefix entry in net data with mDp = 1 and mSlaac=0 is not a valid OMR prefix, while mDp = 1 and mSlaac=1 is valid if the other conditions are also met. (So @abtink 's choice)

I don't think 9.6.1.2 justifies such behavior. Instead, 9.6.1.2 implies that Domain Prefix is considered as OMR because otherwise PBBR can't add Domain Prefix when there is already an OMR prefix in the network, which conflicts with PBBR election behaviors. 
5.23.2.1 also suggests all BBRs to add Domain Prefix which conflicts with how OMR prefix are added and elected. 

Maybe it's most inline with the spec to not treat Domain Prefix as OMR?

Reply:
The current 1.3.0 spec text in 9.6.1.2 ("For the purpose of detection of OMR prefixes ... ") defines what a BR needs to check, to determine if there is an OMR prefix already or not. There's no check on P_dp there which is why I said that not checking on P_dp would be closest aligned to the current spec text.  I don't see a clear benefit yet for checking on P_dp here. 

If the goal is to have DUA addresses and OMR address as 2 completely separate, orthogonal features then indeed it would be best to consider a Domain Prefix as "not suitable as OMR prefix" since the OMR feature would then need to work even if Domain Prefix is configured with P_slaac=1.   But that wasn't our goal (so far).

Reply:
Thank guys! I agree all solutions have some pros and cons and there is no clear winner here IMO.

On on hand, I assume, if we have DUA enabled then it is somewhat of a managed network and we then don't want the auto-generation of OMR and sending of RA msgs. So it may make sense for installer to disable the `RoutingManager` feature asa a whole (on any BR in the installation). So it may not matter if we include or exclude the prefixes `mDp` as OMR?

I agree current PR is most aligned with the current text in spec (allow `mDp=1` and `mSlaac=1` as OMR but don't include it in RA msg) but then if I understand the point from @simonlingoogle is if we have 1.1 devices even if they can assign themselves an IPv6 address using SLAAC, it may not be reachable from outside Thread network (since we don't send it in RA and rely on dua proxy for this)... So it will not practically work.

Reply:
Either solution works for me.

Reply:
I think accepting domain prefix as OMR is fine for now, given that networks using domain prefixes are typically managed. We should revisit this in the spec if this turns out to be an issue.